### PR TITLE
enabling logit scaling

### PR DESCRIPTION
**Pull request by vchiley**
enables scaling output logits; similar to a softmax "temperature"
the [PaLM paper](https://arxiv.org/abs/2204.02311) uses scale 1/sqrt(cfg.d_model) (see Sec 5 Weight initialization)

Implantation notes: 
[F.cross_entropy](https://pytorch.org/docs/stable/generated/torch.nn.functional.cross_entropy.html#torch-nn-functional-cross-entropy) does not enable temperature.
Since we'd have to add it to the model anyway, I just made it a buffer so that it'd be part of the model state_dict.

### CODE REVIEW

File: examples/llm/src/models/mosaic_gpt.py [hunk 1]
```diff
@@ -92,6 +92,13 @@ def __init__(self, cfg: DictConfig):
         self.transformer.update(
             {'ln_f': nn.LayerNorm(cfg.d_model, device=cfg.init_device)})
 
+        # enables scaling output logits; similar to a softmax "temperature"
+        # PaLM paper uses scale 1/sqrt(cfg.d_model)
+        if cfg.get('logit_scale'):
+            self.register_buffer('logit_scale', torch.Tensor([cfg.logit_scale]))
+        else:
+            self.register_buffer('logit_scale', None)
+
```
Comment (abhi-mosaic) [inline examples/llm/src/models/mosaic_gpt.py:101]: From your comment I guess this is in order to save `logit_scale` in the model checkpoint, but it could also just live as part of the `model_cfg` and then we would just use a float rather than a Tensor?

Thinking about this in relation to `alibi_bias` or `softmax_scale` or other args like that, should we just keep it as a `cfg` float for now? I agree we need a better solution for saving model configs colocated with the checkpoint weights, but not sure if we should do a mix of solutions.

Comment (vchiley) [reply examples/llm/src/models/mosaic_gpt.py:101]: ok I can leave it as just part of `self.cfg`. 
I was just trying to be thorough.

**Review by samhavens — APPROVED** [review]
LGTM
